### PR TITLE
Center searchbox vertically

### PR DESCRIPTION
**Pull request by ericschdt**
This patch centers the search box vertically and aligns it with the logo.

### CODE REVIEW

**Comment by horazont** [comment]
Thanks for your PR.

It is not clear to me what you mean by "aligns it with the logo". In the current design, the baseline of the search input is aligned with the baseline of the h1 text and the bottom of the logo.

**Before**

Screen Shot 2019-10-06 at 13 46 13

In your proposed design, the logo and the search box are not aligned at all with respect to each other, at least on my device (Firefox on Debian). See the screenshot below:

**After**

Screen Shot 2019-10-06 at 13 45 34

Maybe I just don’t see it, but I don’t think your changes improve alignment with the logo.

Other than that, I’m pretty torn between alignment of the baseline and the slightly slimmer header + the vertical centering and cannot decide between those two.

File: muchopper/web/scss/app.scss [hunk 1]
```diff
@@ -116,6 +116,7 @@ header#main.slim > .searchbox {
 	flex: 1 1 auto;
 	display: flex;
 	flex-direction: column;
+	margin: auto;
```
Comment (horazont) [inline muchopper/web/scss/app.scss:119]: On a techincal note, the proper way to achieve this would be ``align-self: center;``, since this is a flex item.

**Comment by horazont** [comment]
I put this up on [beta.search.jabber.network](https://beta.search.jabber.network/rooms/1) so that other people can take a look, too.

@ericschdt Can you clarify what you mean by alignment with the logo?

**Comment by ericschdt** [comment]
> Can you clarify what you mean by alignment with the logo?

Sorry, I was mistaken, you are right as it does not align it to the logo. However, centering it still looks nicer to me, though this is just a subjective point of view.

> On a techincal note, the proper way to achieve this would be align-self: center;, since this is a flex item.

You always learn something new. :) If you are fine with this PR, shall I change it accordingly or do you take over the task yourself in the end? In that case I may close this PR again.

Or do you have left the space beneath the searchbar for the filter buttons? However, when they are displayed, the header height currently expands anyway.

**Comment by ericschdt** [comment]
With my patch the searchbar jumps a few pixels upwards if the filter checkboxes become visible. This is not ideal.

**Comment by horazont** [comment]
@ericschdt I got a bit more input and made some changes which center the search box vertically on non-search pages and use the same centered margin on other pages. I put the changes live on [beta.search.jabber.network](https://beta.search.jabber.network/rooms/1), what do you think?

**Comment by ericschdt** [comment]
> what do you think?

That my try was way too trivial comparing with [your solution](https://github.com/horazont/muchopper/commit/a0827076a48ba69af925e5f95e950bae2619cf8c). :-P


Seriously, I like it a lot more now than it was before. 
If just the whole banner does not expand so much for the three filter buttons it'd be perfect imho. :)

Nice work anyway! Though, on the downside, it increases the complexity of the layout a lot ofc.

**Comment by horazont** [comment]
> Though, on the downside, it increases the complexity of the layout a lot ofc.

Yeah, it escalated quite a bit… I’m gathering more feedback, but I also tend to like it.

Expansion of the header is not too bad, and also brings attention to the newly added form fields.

**Comment by horazont** [comment]
Finally decided that this is the way to go. Thanks for submitting the initial patch, @ericschdt.

**Pull request merged**